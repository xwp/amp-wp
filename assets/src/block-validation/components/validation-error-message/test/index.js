/**
 * External dependencies
 */
import { render } from 'enzyme';

/**
 * Internal dependencies
 */
import ValidationErrorMessage from '../';

describe( 'ValidationErrorMessage', () => {
	it( 'renders an error with a custom message', () => {
		const errorMessage = render( <ValidationErrorMessage message="Hello World" /> );
		expect( errorMessage ).toMatchSnapshot();
	} );

	it( 'renders an error for an invalid element', () => {
		const errorMessage = render( <ValidationErrorMessage code="DISALLOWED_TAG" node_name="foo" /> );
		expect( errorMessage ).toMatchSnapshot();
	} );

	it( 'renders an error for an invalid attribute', () => {
		const errorMessage = render( <ValidationErrorMessage code="DISALLOWED_ATTR" node_name="bar" /> );
		expect( errorMessage ).toMatchSnapshot();
	} );

	it( 'renders an error for an invalid attribute with parent node', () => {
		const errorMessage = render( <ValidationErrorMessage code="DISALLOWED_ATTR" node_name="bar" parent_name="baz" /> );
		expect( errorMessage ).toMatchSnapshot();
	} );

	it( 'renders an error for a custom error code', () => {
		const errorMessage = render( <ValidationErrorMessage code="some_other_error" /> );
		expect( errorMessage ).toMatchSnapshot();
	} );

	it( 'renders an error for an unknown error code', () => {
		const errorMessage = render( <ValidationErrorMessage /> );
		expect( errorMessage ).toMatchSnapshot();
	} );
} );
