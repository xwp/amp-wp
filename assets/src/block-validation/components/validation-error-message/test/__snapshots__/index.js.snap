// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationErrorMessage renders an error for a custom error code 1`] = `
Array [
  "Error code: ",
  <code>
    some_other_error
  </code>,
]
`;

exports[`ValidationErrorMessage renders an error for an invalid attribute 1`] = `
Array [
  "Invalid attribute: ",
  <code>
    bar
  </code>,
]
`;

exports[`ValidationErrorMessage renders an error for an invalid attribute with parent node 1`] = `
Array [
  "Invalid attribute: ",
  <code>
    baz[bar]
  </code>,
]
`;

exports[`ValidationErrorMessage renders an error for an invalid element 1`] = `
Array [
  "Invalid element: ",
  <code>
    foo
  </code>,
]
`;

exports[`ValidationErrorMessage renders an error for an unknown error code 1`] = `
Array [
  "Error code: ",
  <code>
    unknown
  </code>,
]
`;

exports[`ValidationErrorMessage renders an error with a custom message 1`] = `null`;
